addons/website_crm_score/controllers/main.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+from openerp import addons, http, SUPERUSER_ID, fields
+from openerp.http import request
+
+
+class PageController(addons.website.controllers.main.Website):
+
+    @http.route('/page/<page:page>', auth="public", website=True)
+    def page(self, page, **opt):
+        response = super(PageController, self).page(page, **opt)
+        # duplication of ir_http.py
+
+        if getattr(response, 'status_code', 0) == 200:
+            try:
+                view = request.website.get_template(page)
+            except:
+                pass  # view not found
+            else:
+                if view.track:  # avoid tracking redirected page
+                    cr, uid, context = request.cr, request.uid, request.context
+                    lead_id = request.env["crm.lead"].decode(request)
+                    url = request.httprequest.url
+                    vals = {'lead_id': lead_id, 'user_id': request.session.get('uid'), 'url': url}
+
+                    if not lead_id or request.registry['website.crm.pageview'].create_pageview(cr, uid, vals, context=context):
+                        # create_pageview failed
+                        response.delete_cookie('lead_id')
+                        request.session.setdefault('pages_viewed', {})[url] = fields.Datetime.now()
+                        request.session.modified = True
+
+        return response
+
+
+class ContactController(addons.website_crm.controllers.main.contactus):
+
+    @http.route(['/crm/contactus'], type='http', auth="public", website=True)
+    def contactus(self, **kwargs):
+        response = super(ContactController, self).contactus(**kwargs)
+        # the cookie is written here because the response is not available in the create_lead function
+        if '_values' in response.qcontext:  # contactus error : fields validation not passed
+            # NOTE: this requires for the lead_id to be added to values in the parent controller
+            lead_id = response.qcontext.get('_values').get('lead_id')
+            if lead_id:  # a new lead has been created
+                lead_model = request.registry['crm.lead']
+                # sign the lead_id
+                sign = lead_model.encode(lead_id)
+                response.set_cookie('lead_id', sign, domain=lead_model.get_score_domain_cookies())
```

Comment:
Because we want to keep the information more than 1 week...
If the lead come back 2 month later, we want to add the information in the first lead created and don't create a new one.